lbrynet/daemon/Daemon.py

```diff
@@ -1361,64 +1361,75 @@ def jsonrpc_account_set(
         return result
 
     @requires(WALLET_COMPONENT)
-    @defer.inlineCallbacks
-    def jsonrpc_account_unlock(self, password):
+    def jsonrpc_account_unlock(self, password, account_id=None):
         """
         Unlock an encrypted account
 
         Usage:
-            account_unlock (<password> | --password=<password>)
+            account_unlock (<password> | --password=<password>) [<account_id> | --account_id=<account_id>]
 
         Options:
-            --password=<password> : (str) password for unlocking wallet
+            --account_id=<account_id>        : (str) id for the account to unlock
 
         Returns:
             (bool) true if account is unlocked, otherwise false
         """
 
-        # the check_locked() in the if statement is needed because that is what sets
-        # the wallet_unlocked_d deferred ¯\_(ツ)_/¯
-        if not self.wallet_manager.check_locked():
-            d = self.wallet_manager.wallet_unlocked_d
-            d.callback(password)
-            result = yield d
-        else:
-            result = True
-        response = yield self._render_response(result)
-        defer.returnValue(response)
+        return self.wallet_manager.unlock_wallet(password.encode(), account_id)
+
+    @requires(WALLET_COMPONENT)
+    def jsonrpc_account_lock(self, account_id=None):
+        """
+        Lock an unlocked account
+
+        Usage:
+            account_lock [<account_id> | --account_id=<account_id>]
+
+        Options:
+            --account_id=<account_id>        : (str) id for the account to lock
+
+        Returns:
+            (bool) true if account is locked, otherwise false
+        """
+
+        return self.wallet_manager.lock_account(account_id)
 
     @requires(WALLET_COMPONENT, conditions=[WALLET_IS_UNLOCKED])
-    def jsonrpc_account_decrypt(self):
+    def jsonrpc_account_decrypt(self, account_id=None):
         """
         Decrypt an encrypted account, this will remove the wallet password
 
         Usage:
-            account_decrypt
+            account_decrypt [<account_id> | --account_id=<account_id>]
 
         Options:
-            None
+            --account_id=<account_id>  : (str) id for the account to decrypt
 
         Returns:
             (bool) true if wallet is decrypted, otherwise false
         """
-        return self.wallet_manager.decrypt_wallet()
+
+        return self.wallet_manager.decrypt_account(account_id)
```

Comment:
Convert `account_id` to an actual account using `self.get_account_or_default(account_id)`.